eclipse-theia/theia
# fix: save current background color if not custom

#### What it does

This PR saves the color of the background for electron main window, even if not changed during the session. In the case for example of setting a light theme in dark mode, when the session during which the theme was saved, the `customBackgroundColor` was set and it is saved in the electron state. The second time, when the app is started and closed and no theme has changed, the `customBackgroundColor` has not been set, so its is not saved. The next time the app is started again, the white color is not restored, thus causing a dark, then white background at startup.
In conjonction with the changes to the setBackgroundColor to the main window, that should solves all issues on startup and window colors.

fix: #14428

contributed on behalf of STMicroelectronics

#### How to test

This can be tested on Theia Example Electron, changing its `package.json` not to have the splashscreen, but `showWindowEarly` set to `true`
1. update package.json in  electron example. Replace [splashcreen options ](https://github.com/eclipsesource/theia/blob/1c328a6d0852052e34cf2629602137b22d01a2e8/examples/electron/package.json#L15) by "showWindowEarly" : "true"
2. compile and run electron example 
3. change the theme to dark while in OS white theme. Restart at least twice, check at startup the windows background and try to refresh electron window (ctrl+R on linux and windows). The background color shall remain consistent.
4. You can test also any combination of a light or dark theme on OS dark / light theme. The background shall remain consistent across several restart.
5. You can also test a fresh start, while removing .theia folder and .config/Theia Electron Example or similar folders on windows.

#### Follow-ups

None known

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
